provisioning_templates/snippet/kickstart_ifcfg_generic_interface.erb

```diff
@@ -54,7 +63,11 @@ DEFROUTE=<%= primary %>
 <%=     "VLAN_NAME_TYPE=VLAN_PLUS_VID_NO_PAD" %>
 <%=     "PHYSDEV=#{@interface.attached_to}" %>
 <%-     if @bonding_interfaces.include?(@interface.attached_to) -%>
-<%=       "TYPE=bonding" %>
+<%-       if @host.operatingsystem.family == 'Redhat' && @host.operatingsystem.major.to_i >= 8 -%>
+<%=         "TYPE=Vlan" %>
+<%-       else -%>
+<%=         "TYPE=bonding" %>
+<%-       end -%>
```

Comment:
This is not right, why would TYPE be VLAN on EL8 but Bonding on EL7? Does not make sense to me, explain.

Reply:
When I look into the nmcli source code (and especially the tests), I find this: [expected type for vlan test](https://github.com/NetworkManager/NetworkManager/blob/master/src/settings/plugins/ifcfg-rh/tests/network-scripts/ifcfg-Vlan_test-vlan-interface.cexpected)

Reply:
I tried to dig deeper, but I got lost in the C code. These are more pointers (pun intended):

Commit: [c2f8400e6606918873505280d0edff3577e7d7ab](https://github.com/NetworkManager/NetworkManager/tree/c2f8400e6606918873505280d0edff3577e7d7ab)

[src/settings/plugins/ifcfg-rh/nms-ifcfg-rh-reader.c](https://github.com/NetworkManager/NetworkManager/blob/c2f8400e6606918873505280d0edff3577e7d7ab/src/settings/plugins/ifcfg-rh/nms-ifcfg-rh-reader.c#L357):

```cpp
static NMSetting *
make_connection_setting (const char *file,
                         shvarFile *ifcfg,
                         const char *type,
                         const char *suggested,
                         const char *prefix)
```

[src/settings/plugins/ifcfg-rh/nms-ifcfg-rh-reader.c](https://github.com/NetworkManager/NetworkManager/blob/c2f8400e6606918873505280d0edff3577e7d7ab/src/settings/plugins/ifcfg-rh/nms-ifcfg-rh-reader.c#L392):

```cpp
   g_object_set (s_con,
                 NM_SETTING_CONNECTION_TYPE, type,
                 NM_SETTING_CONNECTION_UUID, uuid,
                 NM_SETTING_CONNECTION_STABLE_ID, svGetValue (ifcfg, "STABLE_ID", &stable_id),
                 NULL);
```

[src/settings/plugins/ifcfg-rh/nms-ifcfg-rh-reader.c](https://github.com/NetworkManager/NetworkManager/blob/c2f8400e6606918873505280d0edff3577e7d7ab/src/settings/plugins/ifcfg-rh/nms-ifcfg-rh-reader.c#L5645):

```cpp
   con_setting = make_connection_setting (file, ifcfg, NM_SETTING_VLAN_SETTING_NAME, NULL, "Vlan");
```

[libnm-core/nm-setting-vlan.h](https://github.com/NetworkManager/NetworkManager/blob/c2f8400e6606918873505280d0edff3577e7d7ab/libnm-core/nm-setting-vlan.h#L25):

```cpp
#define NM_SETTING_VLAN_SETTING_NAME "vlan"
```

Reply:
Finally, this is also reflected in [the documentation](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/networking_guide/sec-configure_802_1q_vlan_tagging_using_the_command_line_tool_nmcli) on using `nmcli` to add a tagged vlan:

```bash
$ nmcli con add type vlan ifname VLAN10 dev eth0 id 10
Connection 'vlan-VLAN10' (37750b4a-8ef5-40e6-be9b-4fb21a4b6d17) successfully added.
```

Reply:
On the explicit "why" question: I'm guessing, because of this part:

```erb
<%-     if ( @host.operatingsystem.family == 'Redhat' && @host.operatingsystem.major.to_i <= 7 ) -%>
<%=       "NM_CONTROLLED=no" %>
<%-     end -%>
```

Basically, we disabled NetworkManager on older versions.

Reply:
My concern is not about "Vlan" vs "VLAN" but about "VLAN" vs "Bonding".